tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/AzurePipelinesTool/AzurePipelinesTool.cs

```diff
@@ -109,34 +109,51 @@ private void Initialize()
     [McpServerTool, Description("Gets details for a pipeline run")]
     public async Task<Build> GetPipelineRun(int buildId, string? _project = null)
     {
-        // _project state changes to the last successful GET to the build api
-        project ??= project ?? "public";
-
         try
         {
-            var build = await buildClient.GetBuildAsync(_project, buildId);
-            project = _project;
-            return build;
-        }
-        catch { }
+            // _project state changes to the last successful GET to the build api
+            project ??= project ?? "public";
 
-        try
+            try
```

Comment:
Should we just enforce that it's done by HandleCommand, perhaps even with the MCPTool base class? And then if you want it custom you can override/wrap.

```
public async Task HandleCommand(ctx)
{
    return base.HandleCommand((ctx) => <my func>
}
```